client/my-sites/checkout/upsell-nudge/professional-email-upsell/index.jsx

```diff
@@ -1,9 +1,225 @@
+	callback();
+};
+
+const ProfessionalEmailFeature = ( { children } ) => {
+	return (
+		<li>
+			<span>{ children }</span>
+			<Gridicon icon="checkmark" size="18" />
+		</li>
+	);
+};
+
+const ProfessionalEmailUpsell = ( {
+	currencyCode,
+	domainName = 'example.com',
+	handleClickAccept,
+	handleClickDecline,
+	productCost,
+	setCartItem = noopWithCallback,
+} ) => {
+	const translate = useTranslate();
+	const productsList = useSelector( getProductsList );
+
+	const [ mailboxData, setMailboxData ] = useState( buildNewTitanMailbox( domainName, false ) );
+	const [ showAllErrors, setShowAllErrors ] = useState( false );
+
+	const [ mailboxFieldTouched, setMailboxFieldTouched ] = useState( false );
+	const [ passwordFieldTouched, setPasswordFieldTouched ] = useState( false );
+
+	const hasMailboxError =
+		( mailboxFieldTouched || showAllErrors ) && null !== mailboxData.mailbox.error;
+	const hasPasswordError =
+		( passwordFieldTouched || showAllErrors ) && null !== mailboxData.password.error;
+
+	const hasBeenValidated = [ mailboxData.mailbox, mailboxData.password ].some(
+		( { error, value } ) => null !== error || '' !== value
+	);
+
+	const optionalMailboxFields = [ 'alternativeEmail', 'name' ];
+
+	const formattedPrice = translate( '{{price/}} /mailbox /month (billed monthly)', {
+		components: {
+			price: <span>{ formatCurrency( productCost ?? 0, currencyCode ) }</span>,
+		},
+		comment: '{{price/}} is the formatted price, e.g. $20',
+	} );
+
+	const isEligibleForFreeTrial = true;
+	const discountBadge = isEligibleForFreeTrial ? (
+		<span>
+			<Badge type="success">{ translate( '3 months free' ) }</Badge>
+		</span>
+	) : null;
+
+	const onMailboxValueChange = ( fieldName, fieldValue ) => {
+		const updatedMailboxData = {
+			...mailboxData,
+			[ fieldName ]: { value: fieldValue, error: null },
+		};
+		const validatedMailboxData = validateMailboxes(
+			[ updatedMailboxData ],
+			optionalMailboxFields
+		).pop();
+		setMailboxData( validatedMailboxData );
+	};
+
+	const handleAddEmail = () => {
+		setShowAllErrors( true );
+
+		const validatedTitanMailboxes = validateMailboxes( [ mailboxData ], optionalMailboxFields );
+
+		const mailboxesAreValid = areAllMailboxesValid(
+			validatedTitanMailboxes,
+			optionalMailboxFields
+		);
+
+		setMailboxData( validatedTitanMailboxes[ 0 ] );
+
+		if ( ! mailboxesAreValid ) {
+			return;
+		}
+
+		const cartItem = titanMailMonthly( {
+			domain: domainName,
+			quantity: validatedTitanMailboxes.length,
+			extra: {
+				email_users: validatedTitanMailboxes.map( transformMailboxForCart ),
+				new_quantity: validatedTitanMailboxes.length,
+			},
+		} );
+
+		setCartItem( fillInSingleCartItemAttributes( cartItem, productsList ), () =>
+			handleClickAccept( 'accept' )
+		);
+	};
 
-const ProfessionalEmailUpsell = () => {
 	return (
-		<div>
-			<span>Professional Email Upsell placeholder</span>
+		<div className="professional-email-upsell__main">
```

Comment:
Do we really need that `professional-email-upsell__main` CSS class? It's not used anywhere.

Reply:
I will remove as part of some other outstanding items that were flagged as feedback on the design post.